src/shogun/neuralnets/NeuralLinearLayer.cpp

```diff
@@ -188,7 +188,7 @@ float64_t CNeuralLinearLayer::compute_error(SGMatrix<float64_t> targets)
 	float64_t sum = 0;
 	int32_t length = m_num_neurons*m_batch_size;
 	for (int32_t i=0; i<length; i++)
-		sum += (targets[i]-m_activations[i])*(targets[i]-m_activations[i]);
```

Comment:
better!
Even better would be to use a single linalg call to compute this, rather than the loop.
The advantage also is that it would be SIMD and therefore faster